File: php/tests/Unit/Advertisements/Advertisement/Domain/ValueObject/DescriptionTest.php
<?php
declare(strict_types=1);

namespace Tests\Demo\App\Unit\Advertisements\Advertisement\Domain\ValueObject;

use Demo\App\Advertisements\Advertisement\Domain\Exceptions\DescriptionEmptyException;
use Demo\App\Advertisements\Advertisement\Domain\Exceptions\DescriptionTooLongException;
use Demo\App\Advertisements\Advertisement\Domain\ValueObjects\Description;
use PHPUnit\Framework\TestCase;

class DescriptionTest extends TestCase
{
    private const string DESCRIPTION = 'description';

    public function testShouldCreateADescription()
    {
        $advertisementId = new Description(self::DESCRIPTION);
        $this->assertEquals(self::DESCRIPTION, $advertisementId->value());
    }

    public function testShouldThrowAnExceptionWhenDescriptionHasMoreThan200Characters()
    {
        $this->expectException(DescriptionTooLongException::class);
        $this->expectExceptionMessage('Description has more than 200 characters: Has 201 characters');
        $randomString = str_repeat('a', 201);
        new Description($randomString);
    }

    public function testShouldThrowAnExceptionWhenDescriptionIsEmpty()
    {
        $this->expectException(DescriptionEmptyException::class);
        $this->expectExceptionMessage('Description empty');
        new Description('');
    }
}

File: kotlin/src/test/kotlin/unit/advertisement/domain/model/value_object/DescriptionTest.kt
package unit.advertisement.domain.model.value_object

import advertisements.advertisement.domain.exceptions.DescriptionTooLongException
import advertisements.advertisement.domain.value_object.Description
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test


class DescriptionTest
{
    companion object {
        private const val VALID_DESCRIPTION = "Description test"
    }

    @Test
    fun testShouldCreateADescription() {
        val description = Description(VALID_DESCRIPTION)

        Assertions.assertEquals(VALID_DESCRIPTION, description.value())
    }

    @Test
    fun testShouldThrowAnExceptionWhenDescriptionIsEmpty() {
        Assertions.assertThrows(advertisements.advertisement.domain.exceptions.DescriptionEmptyException::class.java) {
            Description("")
        }
    }

    @Test
    fun testShouldThrowAnExceptionWhenDescriptionIsTooLong() {
        Assertions.assertThrows(DescriptionTooLongException::class.java) {
            val longDescription = "a".repeat(201)
            Description(longDescription)
        }
    }
}
